src/integration/masternode/application/services/masternode.service.ts

```diff
@@ -139,41 +147,53 @@ export class MasternodeService {
     await this.masternodeRepo.save(masternode);
   }
 
-  async create(id: number): Promise<Masternode> {
+  async designatePreEnabled(id: number, txId: string): Promise<void> {
+    const masternode = await this.masternodeRepo.findOne(id);
+    if (!masternode) throw new NotFoundException('Masternode not found');
+
+    masternode.state = MasternodeState.PRE_ENABLED;
+    masternode.creationHash = txId;
+    masternode.creationDate = new Date();
+
+    await this.masternodeRepo.save(masternode);
+  }
+
+  async enabled(id: number): Promise<void> {
     const masternode = await this.masternodeRepo.findOne(id);
     if (!masternode) throw new NotFoundException('Masternode not found');
     if (masternode.creationHash) throw new ConflictException('Masternode already created');
 
     masternode.state = MasternodeState.ENABLED;
 
-    return await this.masternodeRepo.save(masternode);
+    await this.masternodeRepo.save(masternode);
   }
 
-  async prepareResign(id: number, signature: string): Promise<Masternode> {
+  async designateResigning(id: number): Promise<void> {
     const masternode = await this.masternodeRepo.findOne(id);
     if (!masternode) throw new NotFoundException('Masternode not found');
     if (masternode.state !== MasternodeState.ENABLED) throw new ConflictException('Masternode not yet created');
-    masternode.signatureLiquidityManager = signature;
 
     masternode.state = MasternodeState.RESIGNING;
 
-    return await this.masternodeRepo.save(masternode);
+    await this.masternodeRepo.save(masternode);
   }
 
-  async resign(id: number): Promise<Masternode> {
+  async designatePreResigned(id: number, txId: string): Promise<void> {
     const masternode = await this.masternodeRepo.findOne(id);
     if (!masternode) throw new NotFoundException('Masternode not found');
     if (!masternode.creationHash) throw new ConflictException('Masternode not yet created');
     if (masternode.resignHash) throw new ConflictException('Masternode already resigned');
-    if (masternode.state !== MasternodeState.PRE_RESIGNED)
+    if (masternode.state !== MasternodeState.RESIGNING)
       throw new ConflictException('Masternode resign is not confirmed');
```

Comment:
Remove this check, or fix the error message and add a similar check to every life cycle method.